sycl/doc/design/CommandGraph.md

```diff
@@ -203,3 +203,42 @@ Level Zero:
+
+| UR | OpenCL | Supported |
+| --- | --- | --- |
+| urCommandBufferCreateExp | clCreateCommandBufferKHR | Yes |
+| urCommandBufferRetainExp | clRetainCommandBufferKHR | Yes |
+| urCommandBufferReleaseExp | clReleaseCommandBufferKHR | Yes |
+| urCommandBufferFinalizeExp | clFinalizeCommandBufferKHR | Yes |
+| urCommandBufferAppendKernelLaunchExp | clCommandNDRangeKernelKHR | Yes |
+| urCommandBufferAppendUSMMemcpyExp |  | No |
+| urCommandBufferAppendUSMFillExp |  | No |
+| urCommandBufferAppendMembufferCopyExp | clCommandCopyBufferKHR | Yes |
+| urCommandBufferAppendMemBufferWriteExp |  | No |
+| urCommandBufferAppendMemBufferReadExp |  | No |
+| urCommandBufferAppendMembufferCopyRectExp | clCommandCopyBufferRectKHR | Yes |
+| urCommandBufferAppendMemBufferWriteRectExp |  | No |
+| urCommandBufferAppendMemBufferReadRectExp |  | No |
+| urCommandBufferAppendMemBufferFillExp | clCommandFillBufferKHR | Yes |
+| urCommandBufferEnqueueExp | clEnqueueCommandBufferKHR | Yes |
+|  | clCommandBarrierWithWaitListKHR | No |
+|  | clCommandCopyImageKHR | No |
+|  | clCommandCopyImageToBufferKHR | No |
+|  | clCommandFillImageKHR | No |
+|  | clGetCommandBufferInfoKHR | No |
+
+Many of the OpenCL functions take a `cl_command_queue` parameter which is not present in most of the UR functions. Instead when a new command buffer is created in `urCommandBufferCreateExp` we also create and maintain a new internal `ur_queue_handle_t` with a reference stored inside of the `ur_exp_command_buffer_handle_t_` struct. This internal queue is then used with the various Append functions. The internal queue is retained and released whenever the owning command buffer is retained or released.
+
+With command buffers being an OpenCL extension, each function is accessed by loading a function pointer to its implementation. These are defined in a common header file in the UR OpenCL adapter. The symbols for the functions are however defined in [OpenCL-Headers](https://github.com/KhronosGroup/OpenCL-Headers/blob/main/CL/cl_ext.h#L102-L110) but it is not known at this time what version of the headers will be used in the UR Github CI configuration, so loading the function pointers will be used until this can be verified. A Future peice of work would be replacing the custom defined symbols with the ones from OpenCL-Headers.
+
+The `UR_DEVICE_INFO_EXTENSIONS` enum can be used with `urDeviceGetInfo` to query if a specified device supports OpenCL command buffers. This will append `ur_exp_command_buffer` to a string pointer passed to the function if the extension is supported.
+
+Known implementations of cl_khr_command_buffer:
+- [OneAPI-Construction-Kit](https://github.com/codeplaysoftware/oneapi-construction-kit) (must enable `OCL_EXTENSION_cl_khr_command_buffer` when building)
+- [PoCL](http://portablecl.org/)
+
```

Comment:
Can you link the command-buffer emulation layer as well https://github.com/bashbaug/SimpleOpenCLSamples/tree/main/layers/10_cmdbufemu